### ISSUE TITLE
Using css margin-top prop to modify headers respected by Obsidian but not by export

### ISSUE THREAD

**Issue by ryran** [KosmosisDire/obsidian-webpage-export#340]
**Describe the bug**
Since last night, I've been playing around a bunch more with tweaking my theme via CSS. One thing in particular I wanted to do was tweak the way headings are rendered. I've been able to adjust  `text-align`, `font-weight`, `font-size`, and `color` no problem -- things look good in obsidian and the changes are properly reflected on export.

Once I tried tweaking `margin-top` I ran into a surprise: obsidian's reading mode seems to renders it in a way that makes sense; however, on export the change doesn't seem to show up at all. I'm curious if there's something I'm doing wrong... an easy way to achieve this with CSS... or if it's just a limitation in the way the plugin handles the CSS?

**Operating System**
Fedora Linux

**To Reproduce**
Using a custom snippet, I'm doing:
```css
.cm-header-3,
.markdown-preview-section h3
{
    font-weight: 500 ;
    font-size: 22px ;
    margin-top: 0 !important;
}
```
I tried various px and em values.. nothing seems to cause a change with the output exported via the plugin, so I just kept the example simple with `0`.

**Screenshots**
From left to right, the following screenshots show obsidian reading mode with my theme default, reading mode with the above (`margin-top: 0`) tweak, and finally, what google-chrome renders from [b19.org/junk2](https://b19.org/junk2), which was exported using that same `margin-top: 0` tweak.
```
obsidian theme default |  obsidian margin-top 0 |  exported margin-top 0
```
![h3-theme-default](https://github.com/KosmosisDire/obsidian-webpage-export/assets/1092618/57bb61f9-7aa5-4260-8c00-3741949d676e) ![h3-live-zero](https://github.com/KosmosisDire/obsidian-webpage-export/assets/1092618/9d8efe12-e6b1-433d-8fd3-55a302bbd89f) ![h3-online-exported-zero](https://github.com/KosmosisDire/obsidian-webpage-export/assets/1092618/b923eeed-7761-4085-af90-da5138cf58bd)

### IMPORTANT: Error Log
No errors.

**Comment by KosmosisDire** [comment]
Ths is due to the way I am setting the margin-top in the plugin styles. I can fix this so it reflects the snippet in export.

**Comment by KosmosisDire** [comment]
Let me know if you find any other snippets or modifications that don't transfer to the export

**Comment by ryran** [comment]
> Let me know if you find any other snippets or modifications that don't transfer to the export

Sure! I was experimenting with customizing callouts in various ways. I spent an hour tweaking tons of things in my css and viewing just in obsidian and then when I finally got it right, I did an export and all kinds of things were broken. Eventually, I narrowed most of it down to the following.

This approach (with or without the `&`) works in obsidian but causes very weird issues on export:
```css
.callout[data-callout="red-shield"] {
    --callout-color: 255, 0, 0;
    --callout-icon: lucide-shield-alert;

    & .callout-content {
        color: #ff0000;
        font-style: italic;
        text-indent: 1em;
        --callout-content-padding: 0.5em 0 0 0.5em;
    }
}
```
When I say "very weird issues', I mean, like.. it messes with definitions of other callouts I have elsewhere in the file. Thankfully, it only took me about an hour to debug what was going on. :smile:
When I simplified the nested callout definitions to the following approach, things started looking right both in obsidian and via export:
```css
.callout[data-callout="red-shield"] {
    --callout-color: 255, 0, 0;
    --callout-icon: lucide-shield-alert;
}

.callout[data-callout="red-shield"] .callout-content {
    color: #ff0000;
    font-style: italic;
    text-indent: 1em;
    --callout-content-padding: 0.5em 0 0 0.5em;
}
```

Nathan, you already know from my previous posts that I'm a complete noob to css, but I got the advice for the former (nested) approach in the obsidian discord when I asked how to make things more compact and efficient. If both ways are common approaches in css-world, then maybe your awesome plugin should address this, but please don't worry about it on my account, as the work-around is obviously not a big deal.

**Comment by ryran** [comment]
Found another example of something similar.
With the help of discord, I figured out how to do something fun:

```css
.callout[data-callout="heading"] {
    --callout-color: 189, 126, 95;
    
    & > .callout-title {
        position: relative;
        
        & > .callout-title-inner {
            text-align: center;
            flex-grow: 1;
            padding-right: 2em;
        }
        & > .callout-icon {
            position: absolute;
            display: none;
        }
        & > .callout-fold {
            position: absolute;
            right: 0;
        }
    }
    & > .callout-content {
        font-style: italic;
        color: #bd7e5f;
        text-align: center;
        --callout-content-padding: 0 3em 0 2em;
    }
}
```

In obsidian, I have the following markdown:
```
> [!heading]- In the middle of the 19th century, Bahá’u’lláh (1817-1892)—the “Glory of God”—delivered a new Revelation from God to humanity. …
> For four decades, thousands of verses, letters and books flowed from His pen, amounting to some 100 volumes of Sacred Writings. This collection offers the principal works of Bahá’u’lláh that have been translated into English.[](https://www.bahai.org/library/authoritative-texts/bahaullah/)
```
And obsidian renders it like this:
![obsidian](https://github.com/KosmosisDire/obsidian-webpage-export/assets/1092618/d700f7eb-fad0-410d-a3e2-09ab284e8f11)

After export though, it shows up like this:
![export](https://github.com/KosmosisDire/obsidian-webpage-export/assets/1092618/46229036-b30a-4223-a6bb-7d704631ad61)

**Comment by ryran** [comment]
In other fun news, I started playing with the [Markdown Attributes plugin](https://plugins.javalent.com/attributes), defining custom styles in my snippets and then applying them to markdown content via things like:
```
*Centered subtitle text here*
{.h1-subtitle}
```

And webpage export handles that like a champ!

**Comment by KosmosisDire** [comment]
That symbol `&` is not part of standard CSS which is why it does not work. I am surprised it works in obsidian, I guess obsidian runs a sass preprocessor on the css. Sass is like a variant of CSS that adds useful features like that and then compiles into css.

If obsidian includes a sass preprocessor with it then I should be able to run that on all the css included with the page before exporting it which would allow that syntax. Definetly an interesting idea for the future.

**Comment by ryran** [comment]
> That symbol `&` is not part of standard CSS which is why it does not work. I am surprised it works in obsidian, I guess obsidian runs a sass preprocessor on the css. Sass is like a variant of CSS that adds useful features like that and then compiles into css. But it is not supported by browsers.

Are you suggesting I might be able to fix this just by removing the `&` from my code? Sweet. I'll try it tomorrow.

I'm sure you're a css wizard, so you can take the following comments from folks in discord with a grain of salt. :grin: 

![ampersand1](https://github.com/KosmosisDire/obsidian-webpage-export/assets/1092618/33bb60a3-7a42-4869-83c8-9bd8774d4069)

![ampersand2](https://github.com/KosmosisDire/obsidian-webpage-export/assets/1092618/2b429d70-764a-4896-9d90-c25b79f6a46c)

**Comment by KosmosisDire** [comment]
You can't just remove the `&`. Nested selectors are not supported by standard CSS. `&` is just the symbol used to denote nesting in SASS. You you should never nest the selectors if you want it to work in the export. Keep them all seperate and unnested.

**Comment by ryran** [comment]
> You can't just remove the `&`. Nested selectors are not supported by standard CSS. `&` is just the symbol used to denote nesting in SASS. You you should never nest the selectors if you want it to work in the export. Keep them all seperate and unnested.
> 
> Nothing is wrong with nesting, it just isn't supported by any browsers, it's a feature outside of standard CSS.

Oh wow! Ok then. I'll take your word for it and simplify my code.

**Comment by KosmosisDire** [comment]
Hey, can you verify this is fixed in the beta version 1.8.0-10b?

**Comment by KosmosisDire** [comment]
Nice, I'll close this then
